staging/src/k8s.io/apiserver/pkg/registry/generic/registry/storage_factory.go

```diff
@@ -50,7 +50,7 @@ func StorageWithCacher() generic.StorageDecorator {
 			return s, d, err
 		}
 		if klog.V(5).Enabled() {
-			klog.Infof("Storage caching is enabled for %s", objectTypeToString(newFunc()))
+			klog.InfoS("Storage caching is enabled", "type", objectTypeToString(newFunc()))
```

Comment:
before:
I0121 16:59:03.150391    8631 storage_factory.go:53] Storage caching is enabled for apiVersion=stable.example.com/v1alpha1, kind=MultiVersion
after:
I0121 16:59:03.150405    8631 storage_factory.go:54] "Storage caching is enabled" type="apiVersion=stable.example.com/v1alpha1, kind=MultiVersion"

Reply:
The type of newFunc() might not be K8S object type ,and even if so,  apiVersion and kind can not be get directlt.
```
func objectTypeToString(obj runtime.Object) string {
	// special-case unstructured objects that tell us their apiVersion/kind
	if u, isUnstructured := obj.(*unstructured.Unstructured); isUnstructured {
		if apiVersion, kind := u.GetAPIVersion(), u.GetKind(); len(apiVersion) > 0 && len(kind) > 0 {
			return fmt.Sprintf("apiVersion=%s, kind=%s", apiVersion, kind)
		}
	}
	// otherwise just return the type
	return fmt.Sprintf("%T", obj)
}
```

Reply:
Instead of generating string with `key=value` we should pass them all to klog.InfoS. That way we would change:
```
type="apiVersion=stable.example.com/v1alpha1, kind=MultiVersion"
```
to
```
apiVersion="stable.example.com/v1alpha1" kind="MultiVersion"
```

To do that I would propose to create alternative function to `objectTypeToString` called `objectTypeToArgs` that would return list of strings that are positional arguments so we can call
```
klog.InfoS("Storage caching is enabled", objectTypeArgs(newFunc())...)
```

Implementation
```
func objectTypeToArgs(obj runtime.Object) []string {
	// special-case unstructured objects that tell us their apiVersion/kind
	if u, isUnstructured := obj.(*unstructured.Unstructured); isUnstructured {
		if apiVersion, kind := u.GetAPIVersion(), u.GetKind(); len(apiVersion) > 0 && len(kind) > 0 {
		    return []string{"apiVersion", apiVersion, "kind", kind}
		}
	}
	// otherwise just return the type
    return []string{"type", fmt.Sprintf("%T", obj)}
}
```

Reply:
update